pkg/filters/args.go

```diff
@@ -44,12 +44,28 @@ func (filter *ArgFilter) Filter(eventID events.ID, args []trace.Argument) bool {
 		if !found {
 			return true
 		}
-		// TODO: use type assertion instead of string conversion
+
 		if argName != "syscall" {
-			argVal = fmt.Sprint(argVal)
-		}
-		res := filter.Filter(argVal)
-		if !res {
+			switch argValue := argVal.(type) {
+			case string:
+				if !filter.Filter(argValue) {
+					return false
+				}
+				return true
+			case []string:
+				for _, argV := range argValue {
+					// check all string filters for argName (e.g.: SymbolsLoaded)
+					if !filter.Filter(argV) {
+						return false
+					}
+				}
+				return true
+			default:
+				// TODO: Expand other tracee.Argument types here
+				//       ArgFilter should not be used by unimplemented types
+				return true // never filter unknown types
```

Comment:
My understanding is that we should ALWAYS filter unknown types, but that brakes e2e tests (meaning that those tests are relying in arg types we currently don't filter, or might be filtering wrongly).

Reply:
This will break most behavior right now, let'skeep as is

Reply:
Well I'm only mitigating the issue by skipping object related events from the derivation filtering (for now). Will keep issue https://github.com/aquasecurity/tracee/issues/2486 opened for next milestone.